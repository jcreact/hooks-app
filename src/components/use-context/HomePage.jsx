import React, { useContext } from 'react';

import { UserContext } from './UserContext';

export const HomePage = () => {
    const { user } = useContext(UserContext);

    return (
        <>
            <h1 className="mt-3">Home Page</h1>
            <hr />
            <pre className="font-monospace">
                {JSON.stringify(user, null, 3)}
            </pre>
        </>
    );
};
