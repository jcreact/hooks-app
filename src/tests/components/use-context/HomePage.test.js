
import React from 'react';
import { mount } from 'enzyme';

//Componente
import { HomePage } from '../../../components/use-context/HomePage.jsx';

//Hook
import { UserContext } from '../../../components/use-context/UserContext.js';

describe(`Pruebas en el componente 'HomePage.jsx'.`, () => {

    const user = {
        name: 'Jos√© Palma',
        email: 'test@correo.com'
    };

    const component = mount(
        <UserContext.Provider
            value={{ user }}
        >
            <HomePage />
        </UserContext.Provider>
    );

    test(`Debe renderizar el componente HomePage correctamente.`, () => {
        expect(component).toMatchSnapshot();
        expect(component.find('pre').text()).toBe(JSON.stringify(user, null, 3));
    });

});
