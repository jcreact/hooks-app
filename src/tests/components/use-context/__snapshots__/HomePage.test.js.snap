// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente 'HomePage.jsx'. Debe renderizar el componente HomePage correctamente. 1`] = `
Array [
  <h1
    className="mt-3"
  >
    Home Page
  </h1>,
  <hr />,
  <pre
    className="font-monospace"
  >
    {
   "name": "Jos√© Palma",
   "email": "test@correo.com"
}
  </pre>,
]
`;
